gate-core/src/main/groovy/com/netflix/spinnaker/gate/services/internal/KeelService.java

```diff
@@ -62,6 +63,21 @@
   @POST("/delivery-configs/diff")
   List<Map> diffManifest(@Body DeliveryConfig manifest);
 
+  @GET("/delivery-configs/{name}/environment/{environment}/constraints")
+  List<ConstraintState> getConstraintState(
+      @Path("name") String name,
+      @Path("environment") String environment,
+      @Query("limit") Integer limit);
+
+  @POST("/delivery-configs/{name}/environment/{environment}/constraint/{type}")
+  Response updateConstraintStatus(
+      @Path("name") String name,
+      @Path("environment") String environment,
+      @Path("type") String type,
+      @Query("artifactVersion") String artifactVersion,
```

Comment:
Should this be in a post body instead of the url? the comment could get long

Reply:
(for both the eternal api and calling keel)

Reply:
Comment can only be 1k currently which should be safe, but we can change all of this.